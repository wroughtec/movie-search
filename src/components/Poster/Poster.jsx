// @flow
import React, { Component } from 'react';
import { produce } from 'immer';
import { Spinner } from '../Spinner/Spinner';

type Props = {
  imageBaseUrl: string,
  size: string,
  path: string,
  title: string
};

type State = {
  imgLoaded: boolean,
  imgError: boolean
};

export class Poster extends Component<Props, State> {
  state = { imgLoaded: false, imgError: false };

  handleImageError = () => {
    this.setState(
      produce(this.state, draft => {
        draft.imgError = true;
      })
    );
  };

  handleImageLoad = () => {
    this.setState(
      produce(this.state, draft => {
        draft.imgLoaded = true;
      })
    );
  };

  render() {
    const { imageBaseUrl, size, path, title } = this.props,
      { imgLoaded, imgError } = this.state;

    return (
      <div className="c-poster">
        {!imgLoaded && !imgError && <Spinner />}
        {!imgError ? (
          <img
            onLoad={this.handleImageLoad}
            onError={this.handleImageError}
            className="c-poster__img"
            src={`${imageBaseUrl}${size}${path}`}
            alt={title}
          />
        ) : (
          <div className="c-poster__error">broken_image</div>
        )}
      </div>
    );
  }
}
