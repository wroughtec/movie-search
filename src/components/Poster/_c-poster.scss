.c-poster {
  width: 100%;
}

.c-poster__img {
  display: block;
  max-width: 100%;
}
