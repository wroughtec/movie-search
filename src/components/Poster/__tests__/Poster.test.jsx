import React from 'react';
import { shallow } from 'enzyme';
import { Poster } from '../Poster';

describe('Poster', () => {
  let component, props;

  beforeEach(() => {
    props = {
      title: 'Title',
      imageBaseUrl: 'imageBaseUrl',
      size: 'size',
      path: 'path'
    };
    component = shallow(<Poster {...props} />);
  });
  test('Renders', () => {
    expect(component).toMatchSnapshot();
  });
});
