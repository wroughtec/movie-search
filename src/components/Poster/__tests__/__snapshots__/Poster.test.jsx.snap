// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster Renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Poster
    imageBaseUrl="imageBaseUrl"
    path="path"
    size="size"
    title="Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Spinner />,
        <img
          alt="Title"
          className="c-poster__img"
          onError={[Function]}
          onLoad={[Function]}
          src="imageBaseUrlsizepath"
        />,
      ],
      "className": "c-poster",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Title",
          "className": "c-poster__img",
          "onError": [Function],
          "onLoad": [Function],
          "src": "imageBaseUrlsizepath",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Spinner />,
          <img
            alt="Title"
            className="c-poster__img"
            onError={[Function]}
            onLoad={[Function]}
            src="imageBaseUrlsizepath"
          />,
        ],
        "className": "c-poster",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Title",
            "className": "c-poster__img",
            "onError": [Function],
            "onLoad": [Function],
            "src": "imageBaseUrlsizepath",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
